### PR TITLE
Room UI

### PR DESCRIPTION
**Pull request by DavidPgl**

### CODE REVIEW

File: recursio-client/Global/Server.gd [hunk 1]
```diff
@@ -227,3 +245,59 @@ remote func receive_timeline_picks(player_pick, enemy_pick):
...
+	emit_signal("game_rooms_received", game_room_dic)
+
+
+func send_join_game_room(game_room_id, player_user_name):
+	Logger.info("Send join game room [ID: %s]" %[game_room_id], "room_management")
+	rpc_id(1, "receive_join_game_rooms", game_room_id, player_user_name)
+
+
+remote func receive_game_room_joined(player_id_name_dic, game_room_id):
+	emit_signal("game_room_joined", player_id_name_dic, game_room_id)
```
Comment (AmonShokhinAhmed) [inline recursio-client/Global/Server.gd:281]: Logger output is missing here I think.

File: recursio-server/GameRoom/GameRoomManager.gd [hunk 2]
```diff
@@ -91,22 +104,33 @@ func _is_current_game_room_full() -> bool:
 
 
 func _on_peer_connected(player_id):
```
Comment (AmonShokhinAhmed) [inline recursio-server/GameRoom/GameRoomManager.gd:106]: Can we remove this?

File: recursio-client/UI/Menus/GameRooms/GameRoomSearch.gd [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+onready var _btn_refresh_game_rooms: Button = get_node("Content/TopBar/Btn_Refresh")
+onready var _btn_back: Button = get_node("Content/TopBar/Btn_Back")
+onready var _btn_join_game_room: Button = get_node("Content/BottomBar/Btn_JoinGameRoom")
+
+# Game Room ID - Game Room Name
+var _game_room_dic := {}
+
+
+func _ready():	
+	_room_filter.connect("text_changed", self, "_on_filter_text_changed")
```
Comment (AmonShokhinAhmed) [inline recursio-client/UI/Menus/GameRooms/GameRoomSearch.gd:22]: Add `var _error =... `to supress warning for every connect.

File: recursio-client/UI/Menus/GameRooms/GameRoomCreation.gd [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+signal btn_back_pressed()
+
+onready var _btn_create_room: Button = get_node("Content/BottomBar/Btn_CreateRoom")
+onready var _btn_back: Button = get_node("Content/BottomBar/Btn_Back")
+
+onready var _game_room_name: LineEdit = get_node("Content/VBoxContainer/GameRoomName")
+
+
+func _ready():
+	_btn_create_room.connect("pressed", self, "_on_create_game_room_pressed")
```
Comment (AmonShokhinAhmed) [inline recursio-client/UI/Menus/GameRooms/GameRoomCreation.gd:14]: Add `var _error =... `to supress warning for every connect.

File: recursio-client/UI/Menus/GameRoomUI.gd [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+onready var _game_room_name: Label = get_node("Content/TopBar/GameRoomName")
+onready var _player_list: ItemList = get_node("Content/PlayerList")
+
+onready var _btn_ready: Button = get_node("Content/BottomBar/Btn_Ready")
+onready var _btn_leave: Button = get_node("Content/BottomBar/Btn_Leave")
+
+var _game_room_id
+
+func _ready():
+	_btn_ready.connect("pressed", self, "_on_ready_pressed")
```
Comment (AmonShokhinAhmed) [inline recursio-client/UI/Menus/GameRoomUI.gd:16]: Add `var _error =...` to supress warning for every connect.

File: recursio-client/UI/Menus/StartMenu.gd [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+onready var _game_room_creation: GameRoomCreation = get_node("CenterContainer/GameRoomCreation")
+onready var _game_room_ui: GameRoomUI = get_node("CenterContainer/GameRoom")
+
+onready var _character_manager: CharacterManager = get_node("../CharacterManager")
+onready var _debug_room = Constants.get_value("debug", "debug_room_enabled")
+
+var _in_game_room: bool = false
+
+func _ready():
+	btn_play_tutorial.connect("pressed", self, "_on_play_tutorial")
```
Comment (AmonShokhinAhmed) [inline recursio-client/UI/Menus/StartMenu.gd:20]: Add `var _error =...` to supress warning for every connect.

**Review by AmonShokhinAhmed — COMMENTED** [review]
If I press the Play Online button I think an automatic Refresh should be exectued

If I press ready in a room where another client is already joined, then leave, then rejoin the same room it is not possible to press ready again. But it seems the game gets started on the other client if he presses ready.

It is possible to fake join a room that is already deleted (after all players leave) if refresh is not called afterwards, this breaks the UI.

File: recursio-client/UI/Menus/GameRooms/GameRoomSearch.gd (hunk 3 above)
Comment (DavidPgl) [reply recursio-client/UI/Menus/GameRooms/GameRoomSearch.gd:22]: Done

**Comment by AmonShokhinAhmed** [comment]
Looks very stable now, amazing job! 
Only thing missing now is making it accessible to gamepad (but I think that can be handled when the other menu structures are in place, so that is still a bit in the future).

**Pull request merged**